woocommerce/woocommerce-ios
# Localise dates in Order Details screen

Fixes #854 

Localise Order Update Date, Shipment Date and Order Notes Dates.

Before and after (Order update date and shipment date)
<img src ="https://user-images.githubusercontent.com/2722505/58536691-de816900-8223-11e9-872d-e6dc7cb2f6a7.jpg" width="350"/>

Before and after (notes)
<img src="https://user-images.githubusercontent.com/2722505/58536705-e6d9a400-8223-11e9-8e9c-367d95dee7ba.jpg" width="350"/>

## Changes:
- For the order update date I went with the `relativelyFormattedUpdateString`
- For the other dates, here be dragons! Part of the solution is initialising the DateFormatter declared in `Date+Woo` to use the device's current Locale. I didn't notice any major issues with that, but it is affecting all transformation of dates to strings.
- Also, I changed the formatting of the shipment dates and the notes dates to `long` instead of `medium`. It does not seem to affect English strings, but it makes a huge difference for Spanish strings (as in medium does not even seem localised, long does)

## Testing:
- Checkout the branch.
- Either change the simulator language or edit the WooCommerce scheme, and change the application language and application region to a non-English locale (e.g. Spanish / Spain for, you know, reasons)
<img width="350" alt="Screenshot 2019-05-29 at 3 06 33 PM" src="https://user-images.githubusercontent.com/2722505/58536535-72066a00-8223-11e9-9452-e66fc86868ec.png">
- Navigate to an order, check that the dates are localised

Update release notes:
